### PR TITLE
fix: Fixed requiring dep manager when using location flag

### PR DESCRIPTION
**Pull request by lucashuy**
#### Which issue(s) does this change fix?
#4012 

#### Why is this change necessary?
This changes updates the `--no-interactive` check to consider `--location`. Post change, users can specify `--no-interactive --location` again without having to include `--dependency-manager`.

#### How does it address the issue?
Add an additional check in the if condition to check for `--location`.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/init/command.py [hunk 1]
```diff
@@ -118,9 +118,12 @@ def wrapped(*args, **kwargs):
             package_type = ctx.params.get("package_type")
             base_image = ctx.params.get("base_image")
             dependency_manager = ctx.params.get("dependency_manager")
+            location = ctx.params.get("location")
+
+            java_base_image = base_image and "java" in base_image
 
             # dependency manager is only required for ZIP package type and for java based IMAGE package types
```
Comment (moelasmar) [inline samcli/commands/init/command.py:125]: can you update the comment

**Review by moelasmar — COMMENTED** [review]
can you add an integration test case to cover this update?

File: samcli/commands/init/command.py [hunk 2]
```diff
@@ -118,9 +118,13 @@ def wrapped(*args, **kwargs):
             package_type = ctx.params.get("package_type")
             base_image = ctx.params.get("base_image")
             dependency_manager = ctx.params.get("dependency_manager")
+            location = ctx.params.get("location")
 
-            # dependency manager is only required for ZIP package type and for java based IMAGE package types
-            if package_type == ZIP or (base_image and "java" in base_image):
+            java_base_image = base_image and "java" in base_image
```
Comment (sriram-mv) [inline samcli/commands/init/command.py:123]: Why are we bleeding this information over? especially checking for a string in a base_image?

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: samcli/commands/init/command.py (hunk 2 above)
Comment (lucashuy) [reply samcli/commands/init/command.py:123]: Sorry not sure if I 100% follow, but the base image option is optional and can has a couple of java options (`amazon/java11-base`, `amazon/java8.al2-base`, `amazon/java8-base`) hence the `in` check.

**Review by moelasmar — APPROVED** [review]
It looks good to me. Thanks Lucas

File: samcli/commands/init/command.py (hunk 2 above)
Comment (sriram-mv) [reply samcli/commands/init/command.py:123]: Could you add a fixme/todo with respect to our code organization here, we are looking up an arbitrary string in the base image to determine if its a java image.

**Review by sriram-mv — APPROVED** [review]
Approving with conditional feedback.

**Pull request merged**